### DISCUSSION TITLE
im using gunicorn gthread with flask-socketio and this is error "The WebSocket transport is not available...."

### DISCUSSION THREAD

**Discussion by sodreamon** [miguelgrinberg/Flask-SocketIO#1854] [Q&A]
hi
im using flask +socketio + simple-websocket + gunicorn + gthread + nginx + ubuntu

and this is my error "The WebSocket transport is not available, you must install a WebSocket server that is compatible with your async mode to enable it. See the documentation for details"

im using odroid xu4 sbc includes armv7l
i cant install gevent or eventlet on this computer

so im using gthread
this is flask socketio doc

A third option with Gunicorn is to use the threaded worker, along with the [simple-websocket](https://github.com/miguelgrinberg/simple-websocket) package for WebSocket support. This is a particularly good solution for applications that are CPU heavy or are otherwise incompatible with eventlet and gevent use of green threads. The command to start a threaded web server is:
gunicorn -w 1 --threads 100 module:app

so my question is
"is gunicorn gthread working for socket io in my app????????
can i ignore this error???(The WebSocket transport is not available, you must install a WebSocket server that is compatible with your async mode to enable it. See the documentation for details)

if not, how can i solve this???
i already read almost discussions but every answer was "use gevent, use gevent websocket"

i cant install gevent or eventlet
also my app cant accept gevent or eventlet(Oauth error)

have a nice day bro

**Comment by miguelgrinberg** [comment]
Did you follow the instructions in the documentation with regards to installing simple-websocket? Can I see the output of the `pip freeze` command please?

**Reply by sodreamon** [reply]
this is my 'pip freeze'

amqp==5.1.1
async-timeout==4.0.2
bidict==0.22.0
billiard==3.6.4.0
cachelib==0.9.0
celery==5.2.7
certifi==2022.6.15
charset-normalizer==2.0.12
click==8.1.3
click-didyoumean==0.3.0
click-plugins==1.1.1
click-repl==0.2.0
Deprecated==1.2.13
dnspython==2.2.1
Flask==2.1.1
Flask-Login==0.6.1
Flask-OAuthlib==0.9.6
Flask-SocketIO==5.1.1
Flask-SQLAlchemy==2.5.1
Flask-WTF==1.0.1
greenlet==1.1.2
gunicorn==20.1.0
h11==0.13.0
idna==3.3
itsdangerous==2.1.2
Jinja2==3.1.2
kombu==5.2.4
MarkupSafe==2.1.1
oauthlib==2.1.0
packaging==21.3
prompt-toolkit==3.0.30
pyparsing==3.0.9
python-engineio==4.3.3
python-socketio==5.7.1
pytz==2022.1
redis==4.3.4
requests==2.26.0
requests-oauthlib==1.1.0
simple-websocket==0.7.0
six==1.16.0
SQLAlchemy==1.4.39
urllib3==1.26.10
vine==5.0.0
wcwidth==0.2.5
Werkzeug==2.1.2
wrapt==1.14.1
wsproto==1.1.0
WTForms==3.0.1

**Reply by miguelgrinberg** [reply]
I don't know. I think something isn't quite in the way you describe it, because the warning only comes up when no supported WebSocket server is found, and you do have simple-websocket installed in this case.

If you want to provide a small self-contained example that includes the server code and the requirements file I can give it a try here and debug further.
